src/main/java/org/truffleruby/core/module/ModuleNodes.java

```diff
@@ -1515,7 +1515,8 @@ private RubySymbol addInternalMethod(RubyModule module, String name, InternalMet
 
             final Visibility visibility = DeclarationContext
                     .findVisibilityCheckSelfAndDefaultDefinee(module, callerFrame);
-            module.addMethodConsiderNameVisibility(getContext(), method, visibility, this);
+            module.addMethodConsiderNameVisibility(getContext(), method.withOwner(module).withDeclaringModule(module),
```

Comment:
Since we change `owner` here we should ensure that is the correct semantics, and and so also test `.owner` in the new spec.

Reply:
Sure thing.